helix-dap/src/client.rs

```diff
@@ -323,6 +328,47 @@ impl Client {
         self.caps.as_ref().expect("debugger not yet initialized!")
     }
 
+    /// Create a child debugger that shares the same transport and config as the parent. Then return a reference to this client
+    pub async fn create_child_debugger(
+        &mut self,
+        id: usize,
+    ) -> Result<(&mut Client, UnboundedReceiver<Payload>)> {
+        // Check if the parent debugger has a config
+        let config = match self.config {
+            Some(ref config) => config.clone(),
+            None => {
+                return Err(Error::Other(anyhow!(
+                    "Config is needed on parent debugger before creating child"
+                )))
+            }
+        };
+
+        // Use the same config as the parent to start a new child debugger
+        let result = Client::process(
```

Comment:
For now this is hardcoded to create a new process, but some DAP servers support multiplexing clients to the same server, in future we may want to use TCP if the server sends an `attach` request.